chainstate/storage/src/internal/version.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use serialization::{Decode, Encode};
+
+#[derive(Debug, Encode, Decode, Clone, Copy, Eq, PartialEq)]
+pub struct ChainstateStorageVersion(u32);
```

Comment:
I've seen many discussions here about what to do about this... and I just want to point out that if we want to make it perfect, we can put all the information (version, magic bytes and net type) in one struct. But I don't really care that much. I'm OK with the way things are now as long as it doesn't lead to a crash and the worst thing that can happen is a wipe and resync.